lib/grp.gi

```diff
@@ -4620,6 +4629,23 @@ InstallMethod( MaximalNormalSubgroups, "for simple groups",
               function(G) return [ TrivialSubgroup(G) ]; end);
 
 
+#############################################################################
+##
+#M  MaximalNormalSubgroups( <G> )
+##
+InstallMethod( MaximalNormalSubgroups, "general method selection",
+              [ IsGroup ],
+    function(G)
+
+    if 0 in AbelianInvariants(G) then
+      # (p) is a maximal normal subgroup in Z for every prime p
+      Error("number of maximal normal subgroups is infinity");
```

Comment:
According to https://codecov.io/gh/gap-system/gap/pull/705/compare this is not covered by the tests. Could you please add a test case? E.g. using an abelian FP group with a single generator)